hail/src/main/scala/is/hail/types/physical/PCanonicalString.scala

```diff
@@ -16,16 +18,8 @@ class PCanonicalString(val required: Boolean) extends PString {
   override def byteSize: Long = 8
 
   lazy val binaryFundamentalType: PCanonicalBinary = PCanonicalBinary(required)
```

Comment:
I don't think there's any benefit to the indirection. here, seems like we should just delete `binaryFundamentalType` and let `fundamentalType = PCanonicalBinary(required)`

Reply:
the PString interface has a `binaryFundamentalType: PBinary` abstract val that this needs to implement. Agree that this needs to change, but hard to do right now. We'll fix this as we get rid of fundamental types.